sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/attributes/NoAccessKeyCheck.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.attributes;
+
+import org.sonar.check.Rule;
+
+@Rule(key = "S6846")
+public class NoAccessKeyCheck extends IllegalAttributeCheck {
```

Comment:
I am not sure it's a good idea to extend `IllegalAttributeCheck`. It defines a rule property that `NoAccessKeyCheck` would likely also inherit.

Furthermore, I am not a big fan of having rules extending others. From experience with CDK sprints, abstracting over rule patterns becomes painful in the long run, especially when we need to go back and make some changes that could impact more than one rule unexpectedly.